### PR TITLE
feat!: add 'name' field in event struct

### PR DESCRIPTION
**Pull request by LycrusHamster**
**What type of PR is this?**
BREAKING CHANGE

feature

**What this PR does / why we need it**:
to enhance event to include one new field:
name : String , this indicates the event name in the context

provided applied, this pr will help client to use 'name' field to filter events they interest, instead of try read JSON data to identify JSON data's type

### CODE REVIEW

**Comment by yejiayu** [comment]
plz follow https://www.conventionalcommits.org/en/

File: built-in-services/asset/src/lib.rs [hunk 1]
```diff
@@ -173,7 +173,7 @@ impl<SDK: ServiceSDK> AssetService<SDK> {
             return ServiceResponse::<Asset>::from_error(103, format!("{:?}", e));
         }
         let event_str = event_res.unwrap();
-        ctx.emit_event(event_str);
+        ctx.emit_event("Create".to_owned(), event_str);
```
Comment (homura) [inline built-in-services/asset/src/lib.rs:176]: Consistency with the method name(`create_asset`) will make it clearer.

**Pull request merged**